### ISSUE TITLE
Expose `SummaryTimer` logic so clients can use it when calling `observe`

### ISSUE THREAD

**Issue by checketts** [prometheus/client_java#165]
The built in timer is very convenient for recording measurement when you know the label before hand. However when you don't, like if `responseStatusCode` is a label, and its value isn't known until the request is finishing up, then the user has to fall back to using `observe`

Too often I've seen users fall back to using `millis` or using Guava's stopwatch (which also returns longs, encouraging millis)

It would be great to expose the Timer as a standalone timer I can read elapsed time after the fact so I can get a double that has the right units (meaning seconds).

A simpler approach that would also go along way would be to expose the nanos to seconds conversion as a static method (https://github.com/prometheus/client_java/blob/master/simpleclient/src/main/java/io/prometheus/client/Summary.java#L159)

Then an observation could be as simple as:

```
long startNanos = System.nanoTime();
//Do work

double durationSeconds = Summary.Timer.durationSeconds(startNanos);
//Record observation
```

**Comment by brian-brazil** [comment]
I don't think we should presume that longs are ns, they could be ms if someone used currentTimeMillis. Both of those functions use different bases too (nanoTime isn't guaranteed to use the unix epoch). Thus a different approach is needed.

> if responseStatusCode is a label, and its value isn't known until the request is finishing up, then the user has to fall back to using observe

It's generally recommended not to export latencies in this way, as there's a good chance a human will misinterpret the data.

**Comment by checketts** [comment]
> It's generally recommended not to export latencies in this way, as there's a good chance a human will misinterpret the data.

Are you recommending against recording `responseStatusCode` as a label? I would definitely want to learn more about the recommended way if you have any links or examples.

I agree regarding treating longs as nanos. I prefer the `Timer` class to keep those details hidden.

**Comment by brian-brazil** [comment]
> Are you recommending against recording responseStatusCode as a label?

On latency metrics, yes.

> I would definitely want to learn more about the recommended way if you have any links or examples.

One metric counting requests broken out by response code, another that's just latency.

**Comment by checketts** [comment]
Closed by #173
